packages/nimble-components/src/rich-text-editor/specs/README.md

```diff
@@ -236,6 +242,27 @@ _Props/Attrs_
 -   `markdown` - for retrieving and modifying the markdown value. If the client modifies the markdown value, it will be parsed into a Node using the
     [prosemirror-markdown parser](https://github.com/ProseMirror/prosemirror-markdown/blob/9049cd1ec20540d70352f8a3e8736fb0d1f9ce1b/src/from_markdown.ts#L199).
     The parsed node will then be rendered in the viewer component as rich text.
+-   `external-link` - is a boolean attribute determining whether all absolute links within the viewer component should open in a new tab. By default, all the links will open in the same tab
+    as per the accessibility guidelines on WCAG (links are below). If opening in a new tab, `nimble-icon-up-right-from-square` will be placed right next to all the links
+    in the viewer instance to indicate the link will be opened in a new tab.
+    -   Accessibility guidelines to open link only in new tab when required: <https://www.w3.org/TR/WCAG20-TECHS/G200.html>
+    -   Accessibility guidelines on opening a link in a new tab: <https://www.w3.org/TR/WCAG20-TECHS/G201.html>
+
+_Alternative implementations:_
+
+For comments feature in SLE, there is a requirement to have all links open in a new tab. However, we are also considering an alternative approach where links will open in a new tab only if
```

Comment:
Where is the requirement for having all links open in a new tab coming from? Based on my understanding, that would go against the guidelines you linked right above this paragraph _Accessibility guidelines to open link only in new tab when required: <https://www.w3.org/TR/WCAG20-TECHS/G200.html>_.

Could you also clarify why we'd want some links to have different default behavior than other links? My initial thought is that will be confusing and seemingly inconsistent to the user, but perhaps I'm missing some precedent/guidelines we are following with that.

Reply:
Hi @mollykreis ,

As a part of the Comments requirement, the expectation is to open the links in new tab. Please find the below comment in the Feature - https://dev.azure.com/ni/DevCentral/_backlogs/backlog/ASW%20SystemLink%20Platform/Initiatives/?workitem=2383256

![image](https://github.com/ni/nimble/assets/123383804/852081dd-43e4-4252-99c5-38f3778e24d6)


We observed, in Test Insights, in the grid. the SLE Links are opened in a different tab.
![image](https://github.com/ni/nimble/assets/123383804/c36b7110-34df-4f6b-ab1f-c91d70302f12)

But when we explored how the Azdo PR comments work, we noticed that the links that belong to the same domain are opened in the same tab and the links that belong to the different domain are opened in a different tab
![image](https://github.com/ni/nimble/assets/123383804/79801e56-36de-4d94-a9d6-41a9edd20f2d)

Hence we decided to add this in the `alternative implementation` section to get your suggestions

Reply:
As of fairly recently we have been developing clearer guidelines around when to open links in new tabs based on the WCAG guidelines referenced.

There are two conditions we need to consider:

- From an accessibility perspective: If it is appropriate to open a link in a new tab, how do we show the user the link will open in a new tab. The visual to use has not been defined yet.
- From a security perspective: If the link is "not trusted", how do we show the user the link is not trusted. The visual to use has not been defined yet.

So for the comments feature we have two questions:

1. Is it appropriate for the links to always open in a new tab?
2. Are there concerns about the link being untrusted?

## Is it appropriate for the links to always open in a new tab

The Rich Text Viewer just represents text with links on a page. There is nothing unique about the links in the Rich Text Viewer. Based on the WCAG discussions, the links are on rendered content, there is no implicit behavior of data loss, no explicit switch between security contexts, etc. These are plain links.

So **no**, the comment viewer should not make links always open in a new tab. They should behave like normal links.

The `anchor-target` feature can be useful for cases inside of iframes etc. So I think it is a useful addition generally. **However** it is not a feature that should be used in the comment viewer in SystemLink to always force all links to open in a new window.

## Are there concerns about the link being untrusted

Right now I don't think there are because the comment editor only supports `absolute URLs`. The user can always see very clearly what a link is and make the assessment if it is untrusted. The concern is currently mitigated because we do not support applying URLs to arbitrary text.

**Proposal:** We do not add visuals for opening the url in a new tab or indicating the URL is untrusted as based on the above discussion they are not currently needed. We can optionally add the `anchor-target` attribute as that is a generally useful feature for the rich text viewer but the SystemLink Comment feature will not use that to force all comment links to open in new tabs.

Reply:
@rajsite thanks for your input. 

We have explained this to Josh and rechecked the importance of this requirement. But what we understood is, opening the link in a new tab is an important requirement (as most of the users are not software developers).

Attaching the comment from Josh for your reference - https://dev.azure.com/ni/DevCentral/_workitems/edit/2383256

![image](https://github.com/ni/nimble/assets/123383804/668f7196-efe4-426d-960d-454e775256ab)

Considering the importance of this requirement, can you please guide us on how to proceed further? 

As mentioned in the spec, exposing the `anchor-target` API could be one option. Please let us know if there are any other better ways to achieve this.

I can schedule a meeting with Josh to discuss this if required.

Reply:
Thanks for the follow-up. I added more details to the linked azdo thread to continue the discussion with POs.

Reply:
Summarizing the key decisions made and resolving this particular comment for scope of this PR: 
1. All links will open in same tab (or default link behavior) until further notice. Respective updates has been done on the HLD.
2. An ADR will be published for WCAG accessibility guidelines for links, along with other platform level recommendations on which the scope of link support for the `nimble-rich-text-*` components will be revisited.